octree_web_viewer/src/state.rs

```diff
@@ -0,0 +1,92 @@
+use crate::backend_error::PointsViewerError;
+use point_viewer::octree;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, RwLock};
+
+/// path information for the octrees
+#[derive(Clone)]
+pub struct OctreeKeyParams {
+    /// Location prefix
+    prefix: PathBuf,
+    /// Location suffix
+    suffix: PathBuf,
+}
+
+impl OctreeKeyParams {
+    pub fn get_octree_address(&self, octree_key: impl AsRef<Path>) -> PathBuf {
+        let mut addr = PathBuf::new();
+        addr.push(self.prefix.clone());
+        addr.push(octree_key.as_ref());
+        addr.push(self.suffix.clone());
+        addr
+    }
+}
+
+#[derive(Clone)]
+pub struct AppState {
+    /// LRU Cache for Octrees
+    pub octree_map: Arc<RwLock<HashMap<String, Arc<octree::Octree>>>>,
+    /// information for retieving octree path
+    pub key_params: OctreeKeyParams,
+    /// backward compatibility to input arguments
+    pub init_octree_id: String,
```

Comment:
How about getting rid of this variable and initially creating an octree inside `octree_map` with this id? Then `return_init_id` could look it up inside the hash map (for 0 or more than 1 entries it could give an error)

Reply:
This is easy with an lru but the currently used hash map requires a key to access the item. The only alternative it is returning an iterator that randomly gets me all the values of the hashmap.  I'd like to preserve the right of the client to restart the browser and get deterministically in the same condition the program launched at any time, no matter how many keys were inserted. Can this be subject of a next pr?

Reply:
Ok, let's leave it as it is for now.